mantidproject/mantid
# Add thickness and event slices to table

This PR creates a new column in the batch reduction table in which the sample thickness will be displayed and can be changed. 

This PR also added EventSlices as an option in the options column in the table.

**To test:**
- Get some sans data the loqdemo folder from the ISIS sample data here downloads.mantidproject.org will work.
- Open the new sans GUI, Load MaskFile.txt as the user file and batch_mode_reduction.csv as the batch file.
- Confirm that the thickness column is populated. 
- Change the thickness column and then go to settings->diagnostics. 
- Click on update rows
- Check that in scale the thickness is set to the thickness you have entered and that the thickness_from_file value equals the original value.

- To test the event slicing go back to the runs tab. In the options column type EventSlices='1:1:5,3-7'
- Go to settings->diagnostics and click update rows again
- in the slice settings confirm that end_time = [2,3,4,5,7] and start_time = [1,2,3,4,3]



Fixes #21260 and #21255 . 

**Release Notes** 


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
